### ISSUE TITLE
LC_ALL is ignored , can't set language via env vars

### ISSUE THREAD

**Issue by dreamflow** [microsoft/pyright#7873]
according to the documentation
https://microsoft.github.io/pyright/#/configuration?id=locale-configuration
i've tried to change the language by setting the env-vars

```python
LC_ALL="de"
LC_MESSAGES="de"
LANG="de"
LANGUAGE="de"
```
i've also tried to set all the above env-vars to "es" and then to "fr" ,
but after restarting vscode , the language in which pylance/pyright
displays errors/warnings/infos is always english .

with the "process explorer"(from sysinternals) , i've checked
that the above changed env-vars are what the pylance process
has and sees in its environment .

expected behaviour :
that pylance/pyright use the language set in the env-var LC_ALL .

my environment :
Windows 10 64bit 22H2
Python 3.12.3 [MSC v.1938 64 bit (AMD64)]
vscode 64bit 1.89.0
vscode extension : pylance 2024.5.1 , pyright 1.1.360

pylance log :
[info] Pylance async client (2024.5.1) started with python extension (2024.6.0)
[info] Pylance language server 2024.5.1 (pyright version 1.1.360, commit 4bc8428d) starting
[info] Setting environmentName for service "<default>": "3.12.3 (global)"
[info] Assuming Python version 3.12.3.final.0
[info] Assuming Python platform Windows

https://github.com/microsoft/pyright/issues/7439#issuecomment-1994671587 said that another related issue #7439 was addressed .

**Comment by erictraut** [comment]
This is functionality provided by the pylance team. 

@rchiodo, @debonte, could one of you please transfer this to the pylance-release project and prioritize it according to your triage process? Thanks!

**Comment by rchiodo** [comment]
I believe this is by design if you're running inside of VS code. VS code's locale takes precedence. You can see that here:
https://github.com/microsoft/pyright/blob/7be85982d6fd609f9b26d5ff05b6b72626f1dd65/packages/pyright-internal/src/localization/localize.ts#L130

You have to set the locale to use in VS code using this command here:
![image](https://github.com/microsoft/pyright/assets/19672699/d9d496c8-1ed1-41fa-a339-04c7cb4647d2)

After I change mine to italian, then Pylance errors show up in Italian:
![image](https://github.com/microsoft/pyright/assets/19672699/87b4d893-75a2-4292-98cd-fc23fabdc4dc)

**Comment by rchiodo** [comment]
Maybe we should update the documentation to say those locale environment variables only apply outside of VS code.
